### PR TITLE
Support scientific visualization in graphics created on the frontend

### PR DESCRIPTION
**Pull request by pmconne**
Sunand's application team wants to produce graphics on the frontend from polyfaces with aux data and have the meshes be deformed and/or recolored using AnalysisStyle.
Cleaned up, documented, and promoted AnalysisStyle APIs.
Ported native code for building AuxChannelTable from PolyfaceAuxData.
Fixed various bugs with PolyfaceAuxData.
Added a button to the toolbar displayed by display-test-app when no iModel is loaded, to open a view in which the meshes previously produced by analysis-importer can be displayed and animated as decorations. This will later be converted to a frontend sample by @Josh-Schifter's team.
Deleted analysis-importer.

Incidental changes:
ThematicDisplay.toJSON() omits properties that match the defaults, instead of producing `undefined` (which becomes `null` when serialized).
Clean up bang-happy ThematicDisplayEditor code.
Fix IndexedPolyface.tryTransformInPlace always returning false, and failing to transform PolyfaceAuxData.

### CODE REVIEW

**Review by markschlosseratbentley — APPROVED** [review]

**Review by markschlosseratbentley — APPROVED** [review]
This looks overall good to me - particularly the thematic changes.

**Comment by pmconne** [comment]
> This looks overall good to me - particularly the thematic changes.

Thanks but it's not ready for review yet.

**Comment by markschlosseratbentley** [comment]
> > This looks overall good to me - particularly the thematic changes.
> 
> Thanks but it's not ready for review yet.

@pmconne - Missed the 'draft' status.  Let me know when and I'll take another look.

**Review by mgooding — APPROVED** [review]

**Review by markschlosseratbentley — CHANGES_REQUESTED** [review]
Running into following test build error:
==[ core-full-stack-tests ]======================================[ 65 of 88 ]==
src/frontend/standalone/ViewportChangedEvents.test.ts(236,131): error TS2345: Argument of type '{ inputName: string; }' is not assignable to parameter of type 'AnalysisStyleProps'.
  Object literal may only specify known properties, and 'inputName' does not exist in type 'AnalysisStyleProps'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! core-full-stack-tests@ build:tests: `tsc 1>&2 && npm run webpack:frontend`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the core-full-stack-tests@ build:tests script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Mark.Schlosser\AppData\Roaming\npm-cache\_logs\2021-06-21T18_56_12_547Z-debug.log
Returned error code: 2
"core-full-stack-tests" failed to build.

**Comment by markschlosseratbentley** [comment]
Running into following test build error:
==[ core-full-stack-tests ]======================================[ 65 of 88 ]==
src/frontend/standalone/ViewportChangedEvents.test.ts(236,131): error TS2345: Argument of type '{ inputName: string; }' is not assignable to parameter of type 'AnalysisStyleProps'.
Object literal may only specify known properties, and 'inputName' does not exist in type 'AnalysisStyleProps'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! core-full-stack-tests@ build:tests: tsc 1>&2 && npm run webpack:frontend
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the core-full-stack-tests@ build:tests script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR! C:\Users\Mark.Schlosser\AppData\Roaming\npm-cache_logs\2021-06-21T18_56_12_547Z-debug.log
Returned error code: 2
"core-full-stack-tests" failed to build.

File: core/common/src/AnalysisStyle.ts [hunk 1]
```diff
@@ -8,76 +8,263 @@
 
 import { Range1d, Range1dProps } from "@bentley/geometry-core";
 import { ThematicGradientSettings, ThematicGradientSettingsProps } from "./ThematicDisplay";
+import { Gradient } from "./Gradient";
 
-/** Properties for display of analysis data
```
Comment (markschlosseratbentley) [inline core/common/src/AnalysisStyle.ts:12]: Running into following test build error:
==[ core-full-stack-tests ]======================================[ 65 of 88 ]==
src/frontend/standalone/ViewportChangedEvents.test.ts(236,131): error TS2345: Argument of type '{ inputName: string; }' is not assignable to parameter of type 'AnalysisStyleProps'.
Object literal may only specify known properties, and 'inputName' does not exist in type 'AnalysisStyleProps'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! core-full-stack-tests@ build:tests: tsc 1>&2 && npm run webpack:frontend
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the core-full-stack-tests@ build:tests script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR! C:\Users\Mark.Schlosser\AppData\Roaming\npm-cache_logs\2021-06-21T18_56_12_547Z-debug.log
Returned error code: 2
"core-full-stack-tests" failed to build.

**Review by markschlosseratbentley — APPROVED** [review]

File: core/common/src/AnalysisStyle.ts (hunk 1 above)
Comment (markschlosseratbentley) [reply core/common/src/AnalysisStyle.ts:12]: @pmconne Feel free to resolve this conversation on your own to get this merged in once all in good shape.

**Review by calebmshafer — APPROVED** [review]
Approving rush.json, vscode and pnpm-lock updates.

**Review by EarlinLutz — APPROVED** [review]

**Review by RBBentley — APPROVED** [review]

**Pull request merged**